### PR TITLE
added required field validation in dto

### PR DESCRIPTION
**Pull request by interestIngc**
## Description

Added validation of required json fields in DTOs and implemented unit tests to assert that.
Updated login.html to process the error returned by "login-precheck" endpoint in case the requested user does not exist. Otherwise, if the precheck endpoint returns an error and the returned salt is null, the "login" endpoint would return the json validation error.

### SECTION 1: RESOURCE SERVER (http://localhost:5001/)
- [x] http://localhost:5001/ home page loads
- [x] "LoginAsUser" button takes you to the user login page
- [x] Log in using "tester" and "12341234" should succeed
- [x] Updating the display name should work
- [x] Log out and Register a new user and try to log in with that new user
- [x] "LoginAsClient" button takes you to the client login page
- [x] Log in using "layer8" and "12341234" should work
- [x] You will be able to see your UID and Secret in profile
- [x] Log out and Register a new client and log in using that should work

### SECTION 2: WE'VE GOT POEMS (http://localhost:5173/)
- [x] Log in with 'tester' / '1234'
- [x] Log in anonymously with Layer8
- [x] Clicking "Get Next Poem" loads different poem correctly x 3
- [x] Clicking "Logout" takes you to the login screen
- [x] Clicking "Register" takes you to the registration page
- [x] Registering with a username, password, and profile image is successful
- [x] Logging in with the new username / password succeeds
- [x] Logging in with Layer8 opens the pop-up
- [x] Logging in with the newly registered credentials from Section 1 succeeds
- [x] User chooses to share their new "Username" & "Country" from the Layer8 Resource Server
- [x] Clicking "Get Next Poem" loads different poem correctly x 3
- [x] Clicking "Logout" takes you to the login page
- [x] Log in with 'tester' / '1234', and then with the credentials from Section 1 succeeds
- [x] This time, DO NOT share "display name" or "country"
- [x] Clicking "Get Next Poem" loads different poem correctly x 3

### SECTION 3: IMSHARER (http://localhost:5174/)
- [x] Main page loads
- [x] Upload of image works
- [x] Reload leads to instant reload (demonstrating proper caching)
- [x] Clicking the newly loaded image shows it in a lightbox

### CODE REVIEW

**Comment by ternakkode** [comment]
how we're going to handle the validation error in the frontend side?

File: server/assets-v1/templates/src/pages/user_portal/login.html [hunk 1]
```diff
@@ -125,6 +125,12 @@ <h1 class="font-bold text-[#4F80E1] text-[40px] text-start mb-2">
             }
           );
           const responseOne = await respOne.json();
+
+          if (responseOne.status === false) {
```
Comment (ternakkode) [inline server/assets-v1/templates/src/pages/user_portal/login.html:129]: if (!responseOne.status)

**Review by ternakkode — COMMENTED** [review]
small comment

**Review by huzaifamk — APPROVED** [review]
Looks good, great work @interestIngc

**Comment by interestIngc** [comment]
> how we're going to handle the validation error in the frontend side?

Currently it's handled the same way as other errors, i.e. if the returned status code is not 200 / 201, then the error message is displayed.

**Review by ternakkode — APPROVED** [review]

**Pull request merged**